OptionSelector/CatagoryTableViewController.m

```diff
@@ -0,0 +1,95 @@
+//
+//  CatagoryTableViewController.m
+//  OptionSelector
+//
+//  Created by Jason Wang on 8/14/15.
+//  Copyright (c) 2015 Mike Kavouras. All rights reserved.
+//
+
+#import "CatagoryTableViewController.h"
+#import "CQCategory.h"
+#import "DetailCategoryTableViewController.h"
+
+
+@interface CatagoryTableViewController ()
+
+@end
+
+@implementation CatagoryTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    CQCategory *carSection = [[CQCategory alloc] init];
+    
+    carSection.name = @"Cars";
+    carSection.options = @[@"Hyundai Santa Fe",
+                           @"Jeep Cherokee",
+                           @"Kia Sportage",
+                           @"Mazda CX-5",
+                           @"Honda CR-V"
+                           ];
+    
+    CQCategory *computerSection = [[CQCategory alloc] init];
+    
+    computerSection.name = @"Computers";
+    computerSection.options = @[@"MacBook Air",
+                                @"MacBook Pro",
+                                @"Dell",
+                                @"HP",
+                                @"Chrome"
+                                ];
+    
+    CQCategory *phoneSection = [[CQCategory alloc] init];
+    
+    phoneSection.name = @"Phone";
+    phoneSection.options = @[@"Iphone",
+                             @"Blackberry",
+                             @"Samsung",
+                             @"Sony",
+                             @"Nokia",
+                             @"HTC",
+                             ];
+
+    self.categorySections = @[carSection, computerSection, phoneSection];
+}
+
+- (void)viewDidAppear:(BOOL)animated {
+    [self.tableView reloadData];
```

Comment:
Two comments here. First, when you're overriding methods like `viewDidLoad` or `viewDidAppear:`, you should always call the `super` implementation first:
```objc
[super viewDidAppear:animated];
//...
```
That way, you'll run the code in the superclass's implementation of this method, and then your code.

Second, you probably noticed that after you select an option in the detail view and then go back, there's a delay before any changes are reflected in the master table view. If you wanted the change to be more instantaneous, you could create a `delegate` property on your detail view controller and set it to `self` when preparing to segue from the master to the detail controller, and send a message to the master controller whenever an option was selected indicating that it should reload its table view.